apps/bfd-db-migrator/src/test/java/gov/cms/bfd/migrator/app/MigratorAppIT.java

```diff
@@ -242,21 +289,39 @@ public void testMigrationRunWhenSchemaFileErrorExpectFailedExitCodeAndMigrations
   }
 
   /**
-   * Gets the number of migration scripts by checking the directory they are located in.
+   * Gets the number of migration scripts by checking the directory they are located in and counting
+   * the files.
    *
    * @return the number migration scripts
    */
-  public int getNumMigrationScripts() {
-    String migrationPath = "src/main/resources/db/migration";
-
-    long fileCount = 0;
-    try (Stream<Path> files = Files.list(Paths.get(migrationPath))) {
-      fileCount = files.count();
-    } catch (IOException io) {
-      fail("Could not find file path for migration scripts at: " + migrationPath);
+  public int getNumMigrationScripts() throws IOException {
+
+    int MAX_SEARCH_DEPTH = 5;
+    Path jarFilePath =
+        Files.find(
+                Path.of("target/db-migrator/"),
+                MAX_SEARCH_DEPTH,
+                (path, basicFileAttributes) ->
+                    path.toFile().getName().matches("bfd-model-rif-.*.\\.jar"))
+            .findFirst()
+            .orElse(null);
+    if (jarFilePath == null) {
+      throw new IOException("Could not find jar file for testing num migrations");
     }
 
-    return (int) fileCount;
+    JarFile migrationJar = new JarFile(jarFilePath.toFile());
+    Enumeration<? extends JarEntry> enumeration = migrationJar.entries();
+
+    int fileCount = 0;
+    while (enumeration.hasMoreElements()) {
+      ZipEntry entry = enumeration.nextElement();
+
+      // Check for sql migration scripts
+      if (entry.getName().startsWith("db/migration/") && entry.getName().endsWith(".sql")) {
```

Comment:
I need to test this, but can provide an example of flyway functionality that we currently use. So for any given ".sql" migration file, we may also have a corresponding ".sql.conf" file. I think the test will be fine.

Reply:
Yeah that would end with .conf, so it shouldnt pull those. I was thinking about that as well at some point.